docs/manual.md

```diff
@@ -609,6 +610,168 @@ $ cargo run -- --experimental-encryption \
 ```
 
 
+## Vector search
+
+Turso supports vector search for building workloads such as semantic search, recommendation systems, and similarity matching. Vector embeddings can be stored and queried using specialized functions for distance calculations.
+
+### Vector types
+
+Turso supports both **dense** and **sparse** vector representations:
+
+#### Dense vectors
+
+Dense vectors store a value for every dimension. Turso provides two precision levels:
+
+* **Float32 dense vectors** (`vector32`): 32-bit floating-point values, suitable for most machine learning embeddings (e.g., OpenAI embeddings, sentence transformers). Uses 4 bytes per dimension.
+* **Float64 dense vectors** (`vector64`): 64-bit floating-point values for applications requiring higher precision. Uses 8 bytes per dimension.
+
+Dense vectors are ideal for embeddings from neural networks where most dimensions contain non-zero values.
+
+#### Sparse vectors
+
+Sparse vectors only store non-zero values and their indices, making them memory-efficient for high-dimensional data with many zero values:
+
+* **Float32 sparse vectors** (`vector32_sparse`): Stores only non-zero 32-bit float values along with their dimension indices.
+
+Sparse vectors are ideal for TF-IDF representations, bag-of-words models, and other scenarios where most dimensions are zero.
+
+### Vector functions
+
+#### Creating and converting vectors
+
+**`vector32(value)`**
+
+Converts a text or blob value into a 32-bit dense vector.
+
+```sql
+SELECT vector32('[1.0, 2.0, 3.0]');
+```
+
+**`vector32_sparse(value)`**
+
+Converts a text or blob value into a 32-bit sparse vector.
+
+```sql
+SELECT vector32_sparse('[0.0, 1.5, 0.0, 2.3, 0.0]');
+```
+
+**`vector64(value)`**
+
+Converts a text or blob value into a 64-bit dense vector.
+
+```sql
+SELECT vector64('[1.0, 2.0, 3.0]');
+```
+
+**`vector_extract(blob)`**
+
+Extracts and displays a vector blob as human-readable text.
+
+```sql
+SELECT vector_extract(embedding) FROM documents;
+```
+
+#### Distance functions
+
+Turso provides three distance metrics for measuring vector similarity:
+
+**`vector_distance_cos(v1, v2)`**
+
+Computes the cosine distance between two vectors. Returns a value between 0 (identical direction) and 2 (opposite direction). Cosine distance is computed as `1 - cosine_similarity`.
+
+Cosine distance is ideal for:
+- Text embeddings where magnitude is less important than direction
+- Normalized embeddings from neural networks
```

Comment:
What does it mean? Actually, for normalized embeddings it's better to use dot product (because it will be equivalent but will not require computation of vector norm - since it was normalized and will be equal to 1)